go.mod

```diff
@@ -4,17 +4,6 @@ go 1.16
 
 require (
 	github.com/casbin/casbin/v2 v2.30.2
-	github.com/cncf/udpa/go v0.0.0-20201120205902-5459f2c99403 // indirect
-	github.com/coreos/etcd v3.3.25+incompatible // indirect
-	github.com/coreos/go-semver v0.3.0 // indirect
-	github.com/coreos/go-systemd v0.0.0-20191104093116-d3cd4ed1dbcf // indirect
-	github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f // indirect
-	github.com/envoyproxy/go-control-plane v0.9.5 // indirect
-	github.com/gogo/protobuf v1.3.2 // indirect
-	github.com/golang/protobuf v1.5.2 // indirect
-	github.com/google/uuid v1.2.0 // indirect
-	github.com/stretchr/testify v1.5.1 // indirect
-	go.etcd.io/etcd v3.3.25+incompatible
-	go.uber.org/zap v1.16.0 // indirect
-	google.golang.org/grpc v1.26.0 // indirect; updating this breaks etcd
+	go.etcd.io/etcd/api/v3 v3.5.0
+	go.etcd.io/etcd/client/v3 v3.5.0
```

Comment:
Why there's no indirect dependency?

Reply:
The previous version v3.3.25 wasn't a Go module so its dependencies were listed as indirect. That's why +incompatible as well.
Now v3.5.0 is a Go module and it lists its dependencies properly in its own go.mod. So these are no longer indirect dependencies. Instead these are listed in go.sum.

https://golangbyexample.com/direct-indirect-dependency-module-go/